### ISSUE TITLE
ColorModel.colorToRGB incorrectly removes alpha channel

### ISSUE THREAD

**Issue by CIVITAS-John** [NetLogo/Tortoise#240]
This issue is related to [an issue in Turtle Universe](https://gitee.com/turtle-sim/turtle-universe-issue-tracker/issues/I55N5S). (In Chinese; please use your translator). 

In https://github.com/NetLogo/Tortoise/master/engine/src/main/coffee/engine/core/colormodel.coffee, line 120:

`color.slice(0, 3).map( (c) -> StrictMath.round(c) )`

It is understandable as the `colorToRGB` function might assume the `RGB` format to be the correct output.

However, since it is used in https://github.com/NetLogo/Tortoise/master/engine/src/main/coffee/engine/core/turtle.coffee, line 415:

`@_registerTurtleStamp(@xcor, @ycor, @_size, @_heading, ColorModel.colorToRGB(@_color), @_getShape(), mode.name)`

When the turtle has an RGBA color and tries to stamp, the function will remove the alpha channel which leads to a bug in Turtle Universe. I am surprised, though, that NetLogo Web does not manifest this bug. How did you do it?

**Comment by CIVITAS-John** [comment]
Get it. Hence, we need to change the colormodel.coffee, because it would strip me of the alpha channel which I need to use.

**Comment by TheBizzle** [comment]
No, I don't think so.  You should simply make your own wrapper over `colorToRGB` (like what I linked), and have that wrapper preserve the alpha channel.  There's no need for a change to Tortoise code.  A color's alpha channel is not affected by any calculations that the NetLogo engine does.

**Comment by CIVITAS-John** [comment]
Speaking from the engine perspective, the current behavior of "trimming RGBA to RGB" in places where RGBA colors should be honored may not be correct. So we either want to make a new procedure called colorToRGBA and use it in `turtle.coffee` or we should allow colorToRGBA to keep the 4th channel. 

See the original code I posted:
`@_registerTurtleStamp(@xcor, @ycor, @_size, @_heading, ColorModel.colorToRGB(@_color), @_getShape(), mode.name)`

The fact is, it will correctly convert NL color to RGB; it will correctly carry over RGB; it will mistakenly strip RGBA of its A channel.

I will do a quick hack in TU, but I don't think this should be the expected behavior. And the fix is going to be very quick.

**Comment by TheBizzle** [comment]
All cases that I've tested have shown correct behavior on netlogoweb.org.  However, it is true that stamps and pen lines were dropping transparency information.  I was able to create a test that showed model diff mismatch between desktop and Tortoise in 76005970830fa180bd69bd10f550d3951706e9e2, which I fixed in e8ba529db68d2825b54876182b7b24dc259aa158.

I'm not aware of any outstanding issue here, so I'm closing this ticket.
